tests/test_estimate_pz_task.py

```diff
@@ -0,0 +1,252 @@
+        out_trainz.pzEnsemble.write_to("output_trainz.hdf5")
+        to_delete.append("output_trainz.hdf5")
+        test_out = qp.read("output_trainz.hdf5")
+        assert isinstance(test_out, qp.Ensemble)
+        assert test_out.npdf == 29358
+        for fdel_ in to_delete:
+            os.unlink(fdel_)
+
+    @unittest.skipIf(not IS_S3DF, "Not at S3DF")
+    def test_pz_task_knn_s3df(self):
+        butler = self.makeButler_repo_dc2()
+        model_file_knn_lsst = "model_inform_knn_lsst_wrap.pickle"
+        to_delete = []
+        if not os.path.exists(model_file_knn_lsst):
+            os.system(
+                f"curl -O https://portal.nersc.gov/cfs/lsst/PZ/pz_models/{model_file_knn_lsst}"
+            )
+            to_delete.append(model_file_knn_lsst)
+        pz_model_knn_lsst = PZModel.read(model_file_knn_lsst)
+        task_config_knn = EstimatePZKNNTask.ConfigClass()
+        task_knn = EstimatePZKNNTask(True, config=task_config_knn)
+        task_knn.config.pz_algo.bands = [
+            "mag_u_lsst",
+            "mag_g_lsst",
+            "mag_r_lsst",
+            "mag_i_lsst",
+            "mag_z_lsst",
+            "mag_y_lsst",
+        ]
+        task_knn.config.pz_algo.band_a_env = dict(
+            u=4.81, g=3.64, r=2.70, i=2.06, z=1.58, y=1.31
+        )
+        dd = butler.getDeferred(
+            "objectTable",
+            skymap="DC2",
+            tract=3829,
+            patch=1,
+        ).get(
+            parameters=dict(columns=task_knn.pz_algo.col_names()),
+        )
+        out_knn = task_knn.run(pz_model_knn_lsst, dd)
+        out_knn.pzEnsemble.write_to("output_knn_lsst.hdf5")
+        to_delete.append("output_knn_lsst.hdf5")
+        test_out = qp.read("output_knn_lsst.hdf5")
+        assert isinstance(test_out, qp.Ensemble)
+        assert test_out.npdf == 29358
+        for fdel_ in to_delete:
+            os.unlink(fdel_)
+
+    @unittest.skipIf(CI_HSC_GEN3_DIR is None, "CI_HSC_GEN3 not installed")
+    def test_pz_tasks_ci_hsc(self):
+        model_file_knn_hsc = "model_inform_knn_hsc_wrap.pickle"
+        to_delete = []
+        if not os.path.exists(f"tests/data/{model_file_knn_hsc}"):
+            os.system(
+                f"curl -o tests/data/{model_file_knn_hsc} "
+                f"https://portal.nersc.gov/cfs/lsst/PZ/pz_models/{model_file_knn_hsc}"
+            )
+            to_delete.append(f"tests/data/{model_file_knn_hsc}")
+
+        butler = self.makeButler_ci_hsc(writeable=True)
+        butler.registry.registerDatasetType(self.pzModel_trainz_datasetType)
+        butler.registry.registerDatasetType(self.pzModel_knn_datasetType)
+        butler.registry.registerRun("u/testing/pz_models")
+
+        pzModel_trainz_datasetRef = DatasetRef(
+            self.pzModel_trainz_datasetType,
+            DataCoordinate.from_full_values(
+                self.pzModel_dimension_group,
+                ("HSC",),
+            ),
+            run="u/testing/pz_models",
```

Comment:
Pinging @TallJimbo on whether there's an existing precedent for writing unit tests to a `testing` user collection (and if so, what the username is).

Also, while in practice we don't usually share CI repos between users, it's certainly something you can do and which might break this test if users don't clean up after building.

Reply:
Ok, I put in the cleanup on success.   Happy to just write stuff to the u/$USER..   or u/$USER/pz_testing collection if you prefer.

Reply:
I think there might be some precedent in @mfisherlevine's summit package tests.

Reply:
I couldn't find anything with a quick search, but something like `u/$USER/meas_pz_tests` seems fine.